python/lib/modeldata/dmod/modeldata/data/item_domain_detector.py

```diff
@@ -0,0 +1,211 @@
+from dmod.core.meta_data import DataDomain, DataFormat, DiscreteRestriction, StandardDatasetIndex, TimeRange
+from dmod.core.common.reader import ReadSeeker
+from dmod.core.exception import DmodRuntimeError
+from dmod.core.data_domain_detectors import ItemDataDomainDetector
+from pandas import read_csv as pandas_read_csv
+
+from typing import Optional
+from io import StringIO
+import re
+
+from ..hydrofabric.geopackage_hydrofabric import GeoPackageHydrofabric
+
+# Try to do this if ngen-config package is available
+try:
+    import ngen.config.realization
+    __NGEN_CONFIG_INSTALLED = True
+except ModuleNotFoundError:
+    __NGEN_CONFIG_INSTALLED = False
+
+
+class AorcCsvFileDomainDetector(ItemDataDomainDetector):
+    """
+    Subclass for detecting domains of NextGen regridded per-catchment AORC forcing CSV files.
+
+    Instances must be explicitly or implicitly provided an item name along with the item.  Receiving a :class:`Path`
+    object implicitly satisfies this, with the object's ``name`` property used.  This is required because the applicable
+    catchment for data is not within the data itself; the convention is to include the catchment id as part of the file
+    name.
+    """
+
+    _csv_header: str = "Time,RAINRATE,Q2D,T2D,U2D,V2D,LWDOWN,SWDOWN,PSFC"
```

Comment:
We really shouldn't define that here, since it is defined in the data format itself (although I had to correct/update the format, as the field names have changed over time).  With that done, I stuck in something to compare things to make sure the data looks right.